aten/src/ATen/function_wrapper.py

```diff
@@ -231,6 +231,7 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 ${return_call} TypeDefault::${native_type_method_dispatch}(${native_arguments});
```

Comment:
Do we need the guard for default body case as well?

Reply:
I don't think it is necessary. The reasoning is that TypeDefault implementations don't do variable unwrapping, and expect to take tensors that involve variable.

Reply:
@ezyang I think there is probably some rule/invariant that I'm not aware of. Let me ask a concrete case:

In generate VariableType, it calls at::norm(self, p, dtype) with NoVariable guard: https://bddppq.github.io/codebrowser/pytorch/pytorch/torch/csrc/autograd/generated/VariableType_4.cpp.html#8546
```
  auto tmp = ([&]() {
    at::AutoNonVariableTypeMode non_var_type_mode(true);
    return at::norm(self_, p, dtype);
  })();
```

The corresponding at:: function doesn't seem to have switch: https://bddppq.github.io/codebrowser/pytorch/pytorch/build/aten/src/ATen/Functions.h.html#_ZN2atL4normERKNS_6TensorEN3c106ScalarE
```
static inline Tensor norm(const Tensor & self, c10::optional<Scalar> p, ScalarType dtype) {
#ifdef USE_STATIC_DISPATCH
    return TypeDefault::norm(self, p, dtype);
#else
    static auto table = globalATenDispatch().getOpTable("aten::norm.ScalarOpt_dtype(Tensor self, Scalar? p, *, ScalarType dtype) -> Tensor");
    return table->getOp<Tensor (const Tensor &, c10::optional<Scalar>, ScalarType)>(at::detail::multi_dispatch_tensor_type_set(self))(self, p, dtype);
#endif
}
```

If we don't put AutoNonVariableTypeMode on TypeDefault path then it will be different between the worlds with/without generated VariableType.

Is it guaranteed that this specific version of norm() won't be called from JIT directly? or it's guaranteed that downstream aten code would never CHECK(!is_variable())?

Reply:
You're right! So I guess we do need to set this situation as well. The reason why we can't tell if we're going to call variable or not here, is because it also depends on the contents of derivatives.yaml